tests/rpc/gossamer_utils.go

```diff
@@ -0,0 +1,165 @@
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package rpc
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/ChainSafe/gossamer/dot/rpc/modules"
+)
+
+var (
+	keyList = []string{"alice", "bob", "charlie", "dave", "eve", "fred", "george", "heather"}
+)
+
+// RunGossamer will start a gossamer instance and check if its online and returns CMD, otherwise return err
+func RunGossamer(t *testing.T, nodeNumb int, dataDir string) (*exec.Cmd, error) {
+
+	currentDir, err := os.Getwd()
+	if err != nil {
+		return nil, err
+	}
+
+	gossamerCMD := filepath.Join(currentDir, "../..", "bin/gossamer")
+
+	//var cmd *exec.Cmd
+	//cmd = exec.Command(gossamerCMD, "init",
+	//	"--datadir", dataDir+strconv.Itoa(nodeNumb),
+	//	"--genesis", filepath.Join(currentDir, "../..", "node/gssmr/genesis.json"),
+	//	"--force",
+	//)
+	//
+	////add step for init
+	//t.Log("Going to init gossamer", "cmd", cmd)
+	//err = cmd.Start()
+	//if err != nil {
+	//	t.Error("Could not init gossamer", "err", err)
+	//	return nil, err
+	//}
+
+	//TODO: could we enable genesis file to be configured via args without init?
+	cmd := exec.Command(gossamerCMD, "--port", "700"+strconv.Itoa(nodeNumb),
+		"--key", keyList[nodeNumb],
+		"--datadir", dataDir+strconv.Itoa(nodeNumb),
+		"--rpchost", GOSSAMER_NODE_HOST,
+		"--rpcport", "854"+strconv.Itoa(nodeNumb),
+		"--rpcmods", "system,author,chain",
+		"--key", keyList[nodeNumb],
+		"--config", currentDir+"/config.toml",
+		"--roles", "4",
+		"--rpc",
+	)
+
+	t.Log("Going to execute gossamer", "cmd", cmd)
+	err = cmd.Start()
+	if err != nil {
+		t.Error("Could not execute gossamer cmd", "err", err)
+		return nil, err
+	}
+
+	t.Log("wait few secs for node to come up", "cmd.Process.Pid", cmd.Process.Pid)
+	var started bool
+
+	for i := 0; i < 10; i++ {
+		time.Sleep(3 * time.Second)
+		if err := CheckFunc(t, "http://"+GOSSAMER_NODE_HOST+":854"+strconv.Itoa(nodeNumb)); err == nil {
+			started = true
+			break
+		} else {
+			t.Log("Waiting for Gossamer to start", "err", err)
+		}
+	}
+	if started {
+		t.Log("Gossamer started :D", "cmd.Process.Pid", cmd.Process.Pid)
+	} else {
+		t.Fatal("Gossamer node never managed to start!")
+
+	}
+
+	return cmd, nil
+}
+
+// CheckFunc check if gossamer node is already started
+func CheckFunc(t *testing.T, gossamerHost string) error {
```

Comment:
maybe rename this to something more descriptive? also does it need to be exported?

Reply:
yes it kind of need to be exported. But its inside the tests package, so its not a deliverable and wont be on our godocs.

Reply:
I like this name, really. I am using it like this in other projects and would like to keep it.